pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -46,16 +47,29 @@ const (
 
 const loadBalancerProvisioningStatusActive = "ACTIVE"
 
+var lookupHost = func(host string) (addrs []string, err error) {
+	if net.ParseIP(host) != nil {
+		return []string{host}, nil
+	}
```

Comment:
is it worthy log about the change we are implicitly change from DNS name to IP here?

Reply:
good point, I'm not sure I added into the right place, see https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1738/files#diff-29cb3ac9349ca04685d8b48d0b1bc13a90e76242d0300fea7ed14f4f71ef2eecR74

Reply:
We could directly call `net.LookupHost` as it either returns the IP if it is an IP already or resolves the host otherwise. https://pkg.go.dev/net#LookupHost 

https://go.dev/play/p/06Zss3v3dnI